### PR TITLE
Add rtmp/FlvWriterOptions to support writing audio/video content only

### LINKED ISSUE #1110: Wrong FlvWriter header when audo/video only
**Describe the bug (描述bug)**

FlvWriter直接在header中的媒体标识位中写入[0x05](https://github.com/apache/incubator-brpc/blob/master/src/brpc/rtmp.cpp#L66)(0x04是音频，0x01是视频），忽略了用户可能只需要音频/视频，导致flv的音频/视频流在某些播放器中无法播放。

**To Reproduce (复现方法)**

**Expected behavior (期望行为)**

根据用户输入来写入header。

### PR DESCRIPTION
**Pull request by v1siuol**
Fix #1110

Ref: https://docs.fileformat.com/video/flv/#structure

### CODE REVIEW

File: src/brpc/rtmp.h [hunk 1]
```diff
@@ -402,6 +415,7 @@ class FlvWriter {
 
 private:
     bool _write_header;
+    char _header[9] = { 'F', 'L', 'V', 0x01, 0x05, 0, 0, 0, 0x09 };
```
Comment (zyearn) [inline src/brpc/rtmp.h:418]: 这里不需要把header全存下来，存option就可以，一是因为除了header[4]其余都是写死的，二是option以后可能还会扩展。

File: src/brpc/rtmp.cpp [hunk 2]
```diff
@@ -63,15 +63,19 @@ FlvWriter::FlvWriter(butil::IOBuf* buf)
...
 }
 
-static char g_flv_header[9] = { 'F', 'L', 'V', 0x01, 0x05, 0, 0, 0, 0x09 };
+FlvWriter::FlvWriter(butil::IOBuf* buf, const FlvWriterOptions& options)
+    : _write_header(false), _buf(buf) {
+    const int flags_bit_index = 4;
+    _header[flags_bit_index] = static_cast<uint8_t>(options.flv_content_type);
+}
 
 butil::Status FlvWriter::Write(const RtmpVideoMessage& msg) {
```
Comment (zyearn) [inline src/brpc/rtmp.cpp:72]: char header[9] = { 'F', 'L', 'V', 0x01, _options.flv_content_type, 0, 0, 0, 0x09 };

Comment (v1siuol) [reply src/brpc/rtmp.cpp:70]: Thanks for your feedback. It prompts that "cannot convert that enum type to char in initialization", so I do a type casting in d8003ebe34ecc73bb4d6806b455078afd1cd5b81. Please let me know if you have any concerns.

File: src/brpc/rtmp.cpp [hunk 3]
```diff
@@ -177,6 +185,8 @@ FlvReader::FlvReader(butil::IOBuf* buf)
     : _read_header(false), _buf(buf) {
 }
 
+static char g_flv_header[9] = { 'F', 'L', 'V', 0x01, 0x05, 0, 0, 0, 0x09 };
+
```
Comment (zyearn) [inline src/brpc/rtmp.cpp:189]: 这个g_flv_header是不是可以删除了？下面sizeof的部分可以直接填9，然后在注释里把9的由来（flv header）写清楚就可以了

Comment (v1siuol) [reply src/brpc/rtmp.cpp:189]: 嗯，我觉得可以

**Pull request merged**